### PR TITLE
Scale consistently in TimeGraph

### PR DESCRIPTION
**Pull request by karupayun**
In this PR we are scaling consistently across all elements in TimeGraph, so:

- We are only having one font size (before was impossible because text in
the Timeline was overlapping).
- We are scaling symbol sizes in Buttons accordingly.
- Move a few hardcoded values from Slider to Layout, so they can scale.
- An additional ceiling needed for the horizontal scrollbar to avoid
  having this situation because OpenGl: http://screenshot/3mVGcjPYHgfYuKF

And in addition:
- Rename SliderWidth x SliderWide.
- Refactor a bit Slider class.

Some screenshots:
TimeGraph after zooming-in: http://screenshot/AvYgew2RL5ToNaU
Min size of Scrollbar after zooming-in: http://screenshot/kPY4FfmPBnpeppL
Iterator text is zoomed as well: http://screenshot/M3wLVRPdTJYZmW4

Bugs:
http://b/234358833 -> Scale Timeline - Button - Scrollbar
http://b/200905950 -> Text of iterators doesn't get zoomed.
http://b/230442062 -> Refactor Sliders (although I won't close it since
there is still a bit of cleaning work we can do there).

TODO:
http://b/177200179 -> Scale Qt part (not a priority for now).

Test: Load a capture, see scaling works as. expected.

Edit: Clang11 test was failing because layout was being destroyed. Before we 
were not using layout consistently, so it wasn't failing. I included a fixup.

### CODE REVIEW

File: src/OrbitGl/TimeGraphLayout.h [hunk 1]
```diff
@@ -27,8 +27,10 @@ class TimeGraphLayout {
   float GetTrackContentBottomMargin() const { return track_content_bottom_margin_ * scale_; }
   float GetTrackContentTopMargin() const { return track_content_top_margin_ * scale_; }
   float GetTrackLabelOffsetX() const { return track_label_offset_x_; }
-  float GetSliderWidth() const { return slider_width_; }
-  float GetTimeBarHeight() const { return time_bar_height_; }
+  float GetSliderWide() const { return slider_wide_ * scale_; }
```
Comment (pierricgimmig) [inline src/OrbitGl/TimeGraphLayout.h:30]: I don't get the renaming, "GetSliderWidth()" is perfectly fine. Using the "wide" adjective instead of the "width" noun is weird.

**Review by pierricgimmig — COMMENTED** [review]
Hey Pablo, thanks for this, please see comments, mainly about renaming, but also about the "protected" that was removed.

File: src/OrbitGl/TimeGraphLayout.h (hunk 1 above)
Comment (florian-kuebler) [reply src/OrbitGl/TimeGraphLayout.h:30]: I think it is because it also used to express height. So I suggest "size" is better.

**Review by florian-kuebler — APPROVED** [review]
This is really nice. Thanks a lot! Mainly the same concern about the name "wide", but that is minor.

File: src/OrbitGl/TimeGraphLayout.h (hunk 1 above)
Comment (karupayun) [reply src/OrbitGl/TimeGraphLayout.h:30]: I wanted to suggest a different name because we tend to use width to the x-distance between 2 points (and height to the y_distance), but I will revert it. I don't like size.

This value is:
- The y-size of the horizontal scrollbar
- The x-size of the vertical's one. 

So, I will rollback it.

File: src/OrbitGl/Button.cpp [hunk 2]
```diff
@@ -121,24 +121,26 @@ void Button::DoDraw(PrimitiveAssembler& primitive_assembler, TextRenderer& /*tex
 void Button::DrawSymbol(PrimitiveAssembler& primitive_assembler) {
   const Color kSymbolBaseColor(191, 191, 192, 255);
   const Color kSymbolHighlightColor(255, 255, 255, 255);
-  const float kSymbolPaddingSize = 3.f;
-  const float kSymbolWide = 3.f;
+  // Symbol wide and the padding are related to the size of the button, so they can scale
```
[the remark this answers is not in the document]
Comment (karupayun) [reply src/OrbitGl/Button.cpp:124]: It's because we use width and height for x and y dimensions but width is most-common used to speak about breadth
or broadness (use case here).

File: src/OrbitGl/TimeGraph.cpp [hunk 3]
```diff
@@ -724,12 +721,12 @@ void TimeGraph::UpdateChildrenPosAndContainerSize() {
...
   track_container_->SetPos(timegraph_current_x, timegraph_current_y);
 
-  vertical_slider_->SetWidth(layout_.GetSliderWidth());
+  vertical_slider_->SetWidth(layout_.GetSliderWide());
   vertical_slider_->SetPos(GetWidth() - vertical_slider_->GetWidth(), timegraph_current_y);
 
   timegraph_current_y += track_container_->GetHeight();
   horizontal_slider_->SetWidth(GetWidth() - vertical_slider_->GetWidth());
-  horizontal_slider_->SetPos(timegraph_current_x, timegraph_current_y);
+  horizontal_slider_->SetPos(timegraph_current_x, ceil(timegraph_current_y));
```
[the remark this answers is not in the document]
Comment (karupayun) [reply src/OrbitGl/TimeGraph.cpp]: It's done in the last commit and also I used GetHeight() to make it easier to understand. PTAL.

**Review by florian-kuebler — APPROVED** [review]

File: src/OrbitGl/GlSlider.cpp [hunk 4]
```diff
@@ -309,9 +305,9 @@ void GlHorizontalSlider::DoDraw(PrimitiveAssembler& primitive_assembler,
   // Triangle is 3 pixels smaller than the area - there is a 2 pixel border around the scrollbar,
   // and there is no margin between the border and the triangle to make it as big as possible.
   // On the other side, there is a 1 pixel margin between the triangle and the vertical line
-  float tri_size = slider_resize_pixel_margin_ - 3.f;
-  tri_size = std::min(tri_size, pixel_height_ - tri_size * kHeightFactor - 2.f);
-  const float tri_y_offset = (pixel_height_ - tri_size * kHeightFactor) / 2.f;
+  float tri_size = GetSliderResizeMargin() - 3.f;
```
Comment (pierricgimmig) [inline src/OrbitGl/GlSlider.cpp:308]: nit. with the renaming, we lose information about the units. At other places in the file, "pixels" is used to specify units. I guess everything is in pixels now, so we might want to make the naming consistent and remove "pixels" everywhere, but this is outside the scope of this PR.

**Review by pierricgimmig — APPROVED** [review]
Hi Pablo, thanks for the change, lgtm!

File: src/OrbitGl/GlSlider.cpp (hunk 4 above)
Comment (karupayun) [reply src/OrbitGl/GlSlider.cpp:308]: Yes, I would rather avoid using pixels in CaptureViewElements where we are using world_coordinates instead of Screen's ones, so I agree about removing pixels in this class.
